services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -50,15 +50,32 @@ class HankeController(@Autowired private val hankeService: HankeService) {
      *  TODO: limit call with user information and return only user's own hanke or something?
      *  TODO: do we later on have users who can read all Hanke items?
      */
-    @GetMapping("")
     fun getAllHankeItems(): ResponseEntity<Any> {
         logger.info { "Entering getAllHankeItems" }
         return try {
             val hankeList = hankeService.loadAllHanke()
             ResponseEntity.status(HttpStatus.OK).body(hankeList)
 
-        } catch (e: HankeNotFoundException) {
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(HankeError.HAI1001)
+        } catch (e: Exception) {
+            logger.error(e) {
+                HankeError.HAI1004.toString()
+            }
+            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(HankeError.HAI1004)
+        }
+    }
+
+    @GetMapping
+    fun getHankeList(hankeSearch: HankeSearch?): ResponseEntity<Any> {
+
+        if (hankeSearch == null || hankeSearch.periodBegin == null || hankeSearch.periodEnd == null) {
```

Comment:
Hmm.. jos vain alku tai loppu on annettu, niin kuvittelisin että haluttaisiin hakea kaikki hankkeet jotka ovat jaksolla ikuisuus menneisyydessä...annettu loppu, tai annettu alku...ikuisuus tulevaisuudessa, eli yhdeltä puolelta rajattu haku. Tällä tavalla jos kutsuja jättää toisen pois, niin toiminta vastaa sitä kuin kumpikin arvo olisi jätetty pois.

Toki kutsuja pystyy itse antamaan jonkin pitkällä menneisyydessä/tulevaisuudessa olevan arvon jos niin haluaa nytkin.

Reply:
En tahtonut backend-logiikkaa tuolle luoda ja tämän hetkinen tietokantakutsu vaatii, että molemmissa tulee arvo. Näin ollen ajattelin, että jätetään kutsujan vastuulle miettiä, mikä on "ikuisuus", jonka väliltä data halutaan. Backend tahtoo molemmat arvot tai ei kumpaakaan. Helppohan tuota on sitten jatkokehittää.